### ISSUE TITLE
Support k8s 1.19

### ISSUE THREAD

**Issue by maggieliu** [rancher/rancher#26717]
Support for k8s 1.19

This will also need to include moving from hyperkube to the separate images as described in https://github.com/kubernetes/kubernetes/pull/88676

**Comment by jsolbrig** [comment]
I've encountered an issue with mounting CephFS volumes that is caused by an issue in the hyperkube image that is used for the kubelet. Before submitting an issue either here or to the k8s project, I'm wondering, what would replace the hyperkube container for kubelet?

**Comment by Oats87** [comment]
https://github.com/rancher/hyperkube-base was created to serve as a base for K8s 1.19+.

https://github.com/rancher/hyperkube/tree/v1.19 Is the hyperkube branch for the 1.19 series of Kubernetes.

**Comment by Oats87** [comment]
@jsolbrig We are going to be creating and maintaining our own hyperkube image. The two repositories/v1.19 branch shows how we are doing this, but namely we are simply building an `ubuntu:20.04` based image (with our choice of utilities) and placing the Kubernetes-compiled binaries into the image.

**Comment by Oats87** [comment]
This is ready for test using my fork of KDM, namely
https://github.com/Oats87/kontainer-driver-metadata.git with branch: `k8s-1-19`

1.19 will not work with nft based distributions yet with Flannel and Canal (CentOS 8/RHEL 8) as I am still waiting on a new Flannel release with my implemented iptables-nftables changes. It should work with Calico and Weave.

CentOS 7 / Ubuntu 18.04 / RHEL 7 / etc. will all work with 1.19 with Canal or Flannel (in addition to Calico/Weave)

**Comment by Oats87** [comment]
I have updated my branch of KDM with master build of Flannel that uses the auto-detect functionality. This means that 1.19.x installed from my branch of KDM should be compatible with CentOS 8/RHEL 8 no matter the CNI selected (Calico, Canal, Weave, Flannel)

**Comment by aaronRancher** [comment]
Tested with Rancher server `rancher\rancher:master-502b1839ebeea78f20828706b4e67e907aa9da8f`
Created a custom cluster (1 control, 1 etcd, 3 workers) running Rancher k8s version `v1.19.0-rc.4-rancher1-1`
On Ubuntu 18.04 and 20.04, system workloads are unhealthy. Due to a missing label `node-role.kubernetes.io/worker`
<img width="1305" alt="Screen Shot 2020-08-10 at 12 23 52 PM" src="https://user-images.githubusercontent.com/57409052/89835496-bffe4980-db19-11ea-9f17-ef9617c328c2.png">
Reopening for fix.

**Comment by Oats87** [comment]
Upon initial investigation, the reason for this is due to the fact that the worker nodes in the cluster were not labeled with the correct `node-role.kubernetes.io/worker` label, which caused the workloads to not be able to schedule. Further investigation into this is required, as this does not occur on a `v2.4.5` setup using my branch of KDM.

**Comment by Oats87** [comment]
The following were bumped in this PR for 1.19.0 from 1.18.x:

`rancher/coreos-etcd` from `v3.4.3-rancher1` to `v3.4.13-rancher1`
`rancher/hyperkube` from `v1.18.x-rancherY` to `v1.19.0-rancher1`
`rancher/k8s-dns-kube-dns` from `1.15.2` to `1.15.10` https://github.com/kubernetes/kubernetes/blob/v1.19.0/cluster/addons/dns/kube-dns/kube-dns.yaml.base#L118
`rancher/k8s-dns-dnsmasq-nanny` from `1.15.2` to `1.15.10` https://github.com/kubernetes/kubernetes/blob/v1.19.0/cluster/addons/dns/kube-dns/kube-dns.yaml.base#L174
`rancher/k8s-dns-sidecar` from `1.15.2` to `1.15.10` https://github.com/kubernetes/kubernetes/blob/v1.19.0/cluster/addons/dns/kube-dns/kube-dns.yaml.base#L221
`rancher/k8s-dns-node-cache` from `1.15.7` to `1.15.13` https://github.com/kubernetes/kubernetes/blob/v1.19.0/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml#L143
`rancher/pause` from `3.1` to `3.2` https://github.com/kubernetes/kubernetes/blob/v1.19.0/pkg/kubelet/dockershim/docker_sandbox.go#L43
`rancher/coredns-coredns` from `1.6.9` to `1.7.0` https://github.com/kubernetes/kubernetes/blob/v1.19.0/cmd/kubeadm/app/constants/constants.go#L332
`rancher/cluster-proportional-autoscaler` from `1.7.1` to `1.8.1`
`weaveworks/weave-kube` from `2.6.4` to `2.6.5`
`weaveworks/weave-npc` from `2.6.4` to `2.6.5`

**Comment by Oats87** [comment]
For `v1.19.0-rancher1-1` to `v1.19.1-rancher1-1`, we are bumping:

`rancher/hyperkube` from `v1.19.0-rancher1` to `v1.19.1-rancher1`

`weaveworks/weave-kube` from `2.6.5` to `2.7.0`

`weaveworks/weave-npc` from `2.6.5` to `2.7.0`

Calico components to `v3.16.1`

**Comment by Oats87** [comment]
This is ready to test, by setting the `rke-metadata-config` to
```
{
  "refresh-interval-minutes": "1440",
  "url": "https://raw.githubusercontent.com/Oats87/kontainer-driver-metadata/k8s-1-19-v2.5/data/data.json"
}
```

This must be tested on `master-head`, as it has required `rke` changes.

All network providers for `1.19.x` should be tested, i.e. `canal`, `calico`, `flannel`, and `weave`

**Comment by aaronRancher** [comment]
Logged an issue for Oracle Linux: https://github.com/rancher/rancher/issues/28840

**Comment by aaronRancher** [comment]
Tested with Rancher server `rancher\rancher:master-haed``version  6a6f13c`
k8s `version v1.19.1-rancher1-1`

Deployed clusters and ran automation on the following network configurations:
Calico 
Flanal
Canal
Weave

Deployed clusters in the following OS:
rancheros-v1.5.6-docker-native
RHEl-7.8-Upstream-Docker19.03-Selinux-ON
oracleLinux-7.8-docker-19.03.11 (#28840)
sles15-sp1-docker-19.03.1
ubuntu-20.04-docker-19.03.11

Automation runs did not report any issues.

**Comment by maggieliu** [comment]
Need to merge KDM

**Comment by maggieliu** [comment]
KDM merged.

**Comment by sangeethah** [comment]
On latest master builds - `master-31d7ad327d0287329873868ead67b34f2cda7b01-head` , with `rke-metatdata-config` pointing tot `https://releases.rancher.com/kontainer-driver-metadata/dev-v2.5/data.json` ,` version v1.19.1-rancher1-1` is presented as the default k8s option.

**Comment by sowmyav27** [comment]
**On master-head - commit id: `b1abc415` KDM - `"https://releases.rancher.com/kontainer-driver-metadata/dev-v2.5/data.json"`**

- Deploy a custom cluster - 5 nodes - 1 etcd, 1 control, 3 worker nodes
- Cluster is stuck in `Provisioning` with error `GetDefaultKubernetesServicesOptions: No serviceOptions found for cluster version [v1.19.1-rancher1-1] or cluster major version [v1.19]`

**Comment by Oats87** [comment]
This is fixed in master from `d5ba6012d68281de0869b3430f2ad6bf9204a5d` onwards.

**Comment by sangeethah** [comment]
Tested with rancher server version - `master-0f691dc70f86bbda3d6563af11779300a6191584-head` .
Able to create a K8s cluster with version - `v1.19.1- rancher1-1` successfully.
